### PR TITLE
feat: toAsyncFunction compact with async function

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

**Comment by dead-horse** [comment]
@popomore 看下这个，应该可以解决 egg-schedule 的问题

**Comment by dead-horse** [comment]
这个我先发个版本， #125 再 rebase

**Comment by dead-horse** [comment]
3.17.0

File: test/egg.test.js [hunk 1]
```diff
@@ -392,6 +392,12 @@ describe('test/egg.test.js', () => {
       const wrapped = app.toAsyncFunction(fn);
       return wrapped(true).then(res => assert(res === true));
     });
+
+    it('not translate common function', () => {
+      const fn = arg => Promise.resolve(arg);
+      const wrapped = app.toAsyncFunction(fn);
+      return wrapped(true).then(res => assert(res === true));
```
Comment (fengmk2) [inline test/egg.test.js]: 判断 wrapped === fn 是否同一个 fn 即可。

**Pull request merged**